packages/backend-common/src/database/connectors/postgres.ts

```diff
@@ -60,14 +61,52 @@ export function buildPgDatabaseConfig(
   dbConfig: Config,
   overrides?: Knex.Config,
 ) {
-  return mergeDatabaseConfig(
+  const mergedDatabaseConfig = mergeDatabaseConfig(
     dbConfig.get(),
     {
       connection: getPgConnectionConfig(dbConfig, !!overrides),
       useNullAsDefault: true,
     },
     overrides,
   );
+
+  const azure = dbConfig.getOptional('auth.azure') as any;
+  if (azure && AzureAccessToken.validateCredentials(azure)) {
+    // Initialile Azure AD Authentication.
+    return {
+      ...mergedDatabaseConfig,
+      // Implemeting Azure AD Authentification https://knexjs.org/guide/#configuration-options
+      connection: async () => {
+        const azureAccessToken = new AzureAccessToken(
+          azure.tenantId,
+          azure.clientId,
+          azure.clientSecret,
+        );
+        const {
+          token,
+          expiresOnTimestamp,
+          connectionNumber, // @TODO remove before merge.
+        } = await azureAccessToken.getTokenInfo();
+        console.log(`Creating new connection object ${connectionNumber}`); // @TODO remove before merge.
```

Comment:
Doesn't this functionality belong in the [Postgres connector](https://github.com/backstage/backstage/blob/06c88a8d307caa7eeabc9c461a24cf3a40bac45a/packages/backend-common/src/database/connectors/postgres.ts#L79) or maybe in the [config](https://github.com/backstage/backstage/blob/master/packages/backend-common/src/database/config.ts)?

Reply:
Moved this piece of code to `config.ts`

Reply:
I gonna mark this conversation as resolved. Please reopen if it still relevant :)